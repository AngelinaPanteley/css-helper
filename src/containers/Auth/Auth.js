import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import styles from './Auth.scss';
import * as actions from '../../store/actions/index';

import Auxiliary from '../../hoc/Auxiliary/Auxiliary';
import Spinner from '../../components/UI/Spinner/Spinner';

class Auth extends Component {
  state = {
    email: '',
    password: '',
    isPasswordValid: true,
  }

  onInputChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value,
    })
  }

  validateForm = () => {
    let isValid = true;
    const pass = this.state.password;

    if (pass.length < 6 || pass.length > 12) {
      isValid = false;
      this.setState({ isPasswordValid: isValid });
    }

    return isValid;
  }

  onFormSubmit = (e) => {
    e.preventDefault();

    if (this.validateForm()) {
      this.props.auth(this.state.email, this.state.password, this.props.isAuth);
    }
  }

  render() {
    const linkTitle = this.props.isAuth ? 'Login' : 'Authorization';
    const linkRoute = this.props.isAuth ? '/login' : '/auth';
    const submitValue = this.props.isAuth ? 'Create Account' : 'Log In';

    return (
      <Auxiliary>
        {
          this.props.isLoading
            ?
            <Spinner />
            :
            <form className={styles.Form} onSubmit={this.onFormSubmit}>
              <div className={styles.Form_Item}>
                <input type='email'
                  id='email'
                  name='email'
                  value={this.state.value}
                  onChange={this.onInputChange}
                  required />
                <label htmlFor='email'>
                  Enter Email:
                </label>
              </div>
              <div className={styles.Form_Item}>
                <input type='password'
                  id='password'
                  name='password'
                  value={this.state.value}
                  onChange={this.onInputChange}
                  required />
                <label htmlFor='password'>
                  Enter Password:
                </label>
                {
                  !this.state.isPasswordValid &&
                  <span className={styles.Form_Error}>Password length must be more than 6 and less than 12 symbols.</span>
                }
              </div>
              <div className={styles.Form_Buttons}>
                <Link to={linkRoute}>Switch to {linkTitle}</Link>
                <input type='submit' value={submitValue} />
              </div>
            </form>
        }
      </Auxiliary>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    isLoading: state.auth.loading,
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    auth: (email, password, isAuth) => dispatch(actions.auth(email, password, isAuth)),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Auth);