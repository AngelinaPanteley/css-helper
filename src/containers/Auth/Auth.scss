@import '../../variables';

.Form {
  width: 50%;
  padding-top: 10%;
  margin: auto;

  @include mobile {
    width: 95%;
  }

  &_Item {
    display: flex;
    flex-direction: column-reverse;
    position: relative;

    label {
      color: $color-white;
      font-size: 22px;
      margin-bottom: 10px;
      position: relative;

      &:after {
        content: "";
        background: $color-light-pink;
        transition: all .5s ease-in-out;
        position: absolute;
        bottom: -53px;
        left: 0;
        width: 0;
        height: 3px;
      }
    }

    input:focus + label {
      &:after {
        width: 100%;
        transition: all .5s ease-in-out;
      }
    }

    input {
      height: 40px;
      margin-bottom: 40px;
      font-size: 20px;
      padding: 0 10px;
      color: $color-gray;
      border: none;
      outline: none;
    }
  }

  &_Buttons {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;

    a {
      font-size: 20px;
    }

    input[type='submit'] {
      height: auto;
      border: 0;
      outline: none;
      padding: 10px 80px;
      line-height: 20px;
      background: $color-light-pink;
      color: $color-white;
      transition: $transition;
      font-size: 20px;
      cursor: pointer;

      @include mobile {
        padding: 10px 30px;
      }

      &:hover,
      &:active,
      &:focus {
        background: $color-pink;
        transition: $transition;
      }
    }
  }

  &_Error {
    position: absolute;
    bottom: 15px;
    color: $color-error;
  }
}