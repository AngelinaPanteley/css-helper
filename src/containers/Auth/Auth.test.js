"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import { MemoryRouter, Link } from "react-router-dom";
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import Auth from './Auth.js';
import reducer from '../../store/reducers/index';

configure({ adapter: new Adapter() });
const composeEnhancers = process.env.NODE_ENV === 'development' ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ : null || compose;

const store = createStore(reducer, composeEnhancers(
  applyMiddleware(thunk)
));

describe('>>>Auth', () => {

  //SNAPSHOTS

  const authComponent = <Provider store={store}>
    <MemoryRouter>
      <Auth />
    </MemoryRouter>
  </Provider>;

  const component = renderer.create(authComponent);

  it('renders itself correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });
});
