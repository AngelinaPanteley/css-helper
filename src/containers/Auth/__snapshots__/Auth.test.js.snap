// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Auth renders itself correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <input
      id="email"
      name="email"
      onChange={[Function]}
      required={true}
      type="email"
    />
    <label
      htmlFor="email"
    >
      Enter Email:
    </label>
  </div>
  <div>
    <input
      id="password"
      name="password"
      onChange={[Function]}
      required={true}
      type="password"
    />
    <label
      htmlFor="password"
    >
      Enter Password:
    </label>
  </div>
  <div>
    <a
      href="/auth"
      onClick={[Function]}
    >
      Switch to 
      Authorization
    </a>
    <input
      type="submit"
      value="Log In"
    />
  </div>
</form>
`;
